cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -282,6 +300,112 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 			// imageName := t.ImageName
 			// projectID := gcp.GetProjectID()
 
+		case *target.AzureImageTargetOptions:
+			ctx := context.Background()
+
+			var authorizer autorest.Authorizer
+			if impl.AzureCreds == nil {
+				authorizer, err = auth.NewAuthorizerFromEnvironment()
+				if err != nil {
+					r = append(r, fmt.Errorf("creating an authorizer from environment failed: %v", err))
+					continue
+				}
+			} else {
+				credential := auth.NewClientCredentialsConfig(impl.AzureCreds.ClientID, impl.AzureCreds.ClientSecret, options.TenantID)
+				authorizer, err = credential.Authorizer()
+				if err != nil {
+					r = append(r, fmt.Errorf("creating an azure authorizer failed: %v", err))
+					continue
+				}
+			}
+
+			storageAccountTag := azure.Tag{
+				Name:  "imageBuilderStorageAccount",
+				Value: fmt.Sprintf("location=%s", options.Location),
+			}
+
+			storageAccount, err := azure.GetResourceNameByTag(
+				ctx,
+				authorizer,
+				options.SubscriptionID,
+				options.ResourceGroup,
+				storageAccountTag,
+			)
+			if err != nil {
+				return fmt.Errorf("searching for a storage account failed: %v", err)
+			}
+
+			if storageAccount == "" {
+				storageAccount = generateStorageAccountName()
+
+				err := azure.CreateStorageAccount(
+					ctx,
+					authorizer,
+					options.SubscriptionID,
+					options.ResourceGroup,
+					storageAccount,
+					options.Location,
+					storageAccountTag,
+				)
+				if err != nil {
+					return fmt.Errorf("creating a new storage account failed: %v", err)
+				}
+			}
+
+			storageAccountKey, err := azure.GetStorageAccountKey(
+				authorizer,
+				options.SubscriptionID,
+				options.ResourceGroup,
+				storageAccount,
+			)
+			if err != nil {
+				return fmt.Errorf("retrieving the storage account key failed: %v", err)
+			}
+
+			blobCredentials := azure.Credentials{
+				StorageAccount:   storageAccount,
+				StorageAccessKey: storageAccountKey,
+			}
+
+			storageContainer := "imagebuilder"
+
+			err = azure.CreateStorageContainerIfNotExist(ctx, blobCredentials, storageContainer)
+			if err != nil {
+				return fmt.Errorf("cannot create a storage container: %v", err)
+			}
+
+			blobName := t.ImageName
+			if !strings.HasSuffix(blobName, ".vhd") {
+				blobName += ".vhd"
+			}
+
+			err = azure.UploadImage(
+				blobCredentials,
+				azure.ImageMetadata{
+					ContainerName: storageContainer,
+					ImageName:     blobName,
+				},
+				path.Join(outputDirectory, options.Filename),
+				azure.DefaultUploadThreads,
+			)
+			if err != nil {
+				return fmt.Errorf("uploading the image failed: %v", err)
+			}
+
+			err = azure.RegisterImage(
```

Comment:
For the future: we probably want to have some idea of how long this takes so if it is worth splitting out like what we want to do for AWS. Some logging around the call would do it, I guess.

Reply:
Logging added, fyi it takes about 1 minute to register the image.